### PR TITLE
Adding an example combine container operations and python lightweight components

### PR DESCRIPTION
**Pull request by JohanWork**
This pr adds an example of how to combine container operations and lightweight components. It showcase the versatility of how to setup a pipeline. I think it is an example that helps people that are getting started with kubeflow and showcase that you don't have to pick one of the two options.    

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/772)

### CODE REVIEW

**Comment by JohanWork** [comment]
@NikeNano what are you feedback?

File: README.md [hunk 1]
```diff
@@ -99,6 +99,15 @@ This example covers the following concepts:
 1. How to create pipeline components from python functions in jupyter notebook
 2. How to compile and run a pipeline from jupyter notebook
 
+#### [Ligtheweight component and container operations](./pipelines/ligtheweight-component-and-container-operations)
+Author: [Johan Hansson](https://github.com/JohanWork)
+
+This example covers the following concepts:
+1. How to create pipeline components from python functions and combining it with
+a container operations in jupyter notebook
```
Comment (NikeNano) [inline README.md]: Is this specific for a jupyter notebook? Maybe you could just write ```How to create pipeline components from python functions and combining it with
a container operations ```

Comment (JohanWork) [reply README.md]: No that is true! I will update thnx for the feedback!

**Comment by JohanWork** [comment]
Kind reminder @texasmichelle

**Comment by jlewi** [comment]
@texasmichelle is currently working on other things.

Since this is about pipelines maybe @Ark-kun can review it?

What would the testing story for this example be? How do we ensure it stays up to date and doesn't break?

**Comment by JohanWork** [comment]
I haven't thought about testing to be honest, did know that that existed for the other example piplines  so havent thought about that. Where can I see those test and where are they running @jlewi ?

**Comment by jlewi** [comment]
We have tests to run the example notebooks.
e.g
https://github.com/kubeflow/examples/blob/master/py/kubeflow/examples/notebook_tests/mnist_gcp_test.py

/assign @Ark-kun

**Comment by JohanWork** [comment]
If I add a test is this something we would like to add to the examples @Ark-kun ?

**Comment by jlewi** [comment]
@Ark-kun and @jessiezcc can someone from pipelines please take a look at this?

File: pipelines/ligtheweight-component-and-container-operations/ligtheweight-component-and-container-operations.ipynb [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    "#### Create a python function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.python_component(\n",
```
Comment (Ark-kun) [inline pipelines/ligtheweight-component-and-container-operations/ligtheweight-component-and-container-operations.ipynb:96]: This decorator is being deprecated.
`func_to_container_op` will just use the function docstring.

File: pipelines/ligtheweight-component-and-container-operations/ligtheweight-component-and-container-operations.ipynb [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def echo_op(text1, text2):\n",
+    "    return dsl.ContainerOp(\n",
```
Comment (Ark-kun) [inline pipelines/ligtheweight-component-and-container-operations/ligtheweight-component-and-container-operations.ipynb:142]: We'd like out users to create reusable components in stead of just creating ContainerOp objects.
Creating component.yaml files is just as easy, but you're getting a real component that can be used somewhere else.

**Comment by JohanWork** [comment]
@Ark-kun I have update the notebook with your suggestions, do you have any more feedback otherwise it looks good to me.

**Comment by JohanWork** [comment]
kind reminder @Ark-kun

**Comment by JohanWork** [comment]
kind reminder @Ark-kun and please let me know if this is something we don't want in the examples and I will remove the pr(no hard feelings :) ).

**Comment by JohanWork** [comment]
@Ark-kun

**Pull request closed without merging**